src/glmfit.jl

```diff
@@ -635,11 +668,54 @@ response around the linear predictor. The `:delta` method intervals are symmetri
 the point estimates, but do not respect natural parameter constraints
 (e.g., the lower bound for a probability could be negative).
 """
+
+"""
+    predict(mm::AbstractGLM, newX;
+            offset::FPVector=[],
+            interval::Union{Symbol,Nothing}=nothing, level::Real = 0.95,
+            interval_method::Symbol = :transformation)
+
+Return the predicted response of model `mm` from covariate values `newX` and,
+optionally, an `offset`.
+
+$PREDICT_COMMON
+"""
 function predict(mm::AbstractGLM, newX::AbstractMatrix;
                  offset::FPVector=eltype(newX)[],
                  interval::Union{Symbol,Nothing}=nothing,
                  level::Real=0.95,
                  interval_method=:transformation)
+    r = response(mm)
+    len = size(newX, 1)
+    res = interval === nothing ?
+        similar(r, len) :
+        (prediction=similar(r, len), lower=similar(r, len), upper=similar(r, len))
+    predict!(res, mm, newX,
+             offset=offset, interval=interval, level=level,
+             interval_method=interval_method)
+end
+
+"""
+    predict!(res, mm::AbstractGLM, newX::AbstractMatrix;
+             offset::FPVector=eltype(newX)[],
+             interval::Union{Symbol,Nothing}=nothing, level::Real = 0.95,
+             interval_method::Symbol = :transformation)
+
+Store in `res` the predicted response of model `mm` from covariate values `newX`
+and, optionally, an `offset`. `res` must be a vector with a length equal to the number
+of rows in `newX` if `interval=nothing` (the default), and otherwise a `NamedTuple`
+of vectors with names `prediction`, `lower` and `upper`.
+
+$PREDICT_COMMON
+"""
+function predict!(res::Union{AbstractVector,
+                             NamedTuple{(:prediction, :lower, :upper),
```

Comment:
should we allow `Table.ColumnTable`? Then we have an intermediate method that takes `res::Any` and just called `Table.columntable(res)` to invoke this method making it possible to pass e.g. DataFrames

Reply:
We could do that, but TBH I don't expect people to use this method a lot (if at all). I mainly added it because it makes the design more composable and because I needed it internally. So I'd be inclined to wait and see whether people request it before making the code more complex.